pkg/transform/gorm/models/vmmessages.go

```diff
@@ -5,11 +5,8 @@ import "github.com/filecoin-project/lily/pkg/transform/gorm/types"
 type VmMessage struct {
 	Source  types.DbCID `gorm:"primaryKey"`
 	Cid     types.DbCID `gorm:"primaryKey"`
-	To      types.DbAddr
-	From    types.DbAddr
-	Value   types.DbToken
-	Method  uint64
-	Params  []byte
-	Receipt Receipt `gorm:"embedded"`
+	Message Message     `gorm:"embedded"`
+	Receipt Receipt     `gorm:"embedded"`
```

Comment:
While it is true that VM messages are basically messages, many of the fields such as `GasLimit`, `GasFee`, and `GasPremium` are all `0`. Similarly, VMMessages do not contain a `Signature`, nor do their receipts contain an `EventRoot` (less sure on this, but a hunch). I think it may be more efficient to avoid embedding the `Message` and `Receipt` structures in `VmMessgae` and instead take a subset of the fields that are populated. wdyt?

Reply:
Looking at gorm [Embedded Structs](https://gorm.io/docs/models.html#Embedded-Struct) has me thinking that it's better to have common non-null types between `Message` and `VmMessage` wrapped like:

```
type BaseMessage struct {
  To
  From
  Value
  Method
  Params
}
```

which can then be respectively added as an embedded struct for both `Message` and `VmMessage`. This way it'll be easier to see what they do have in common and what they don't. But this is also not a ⛰️ I'm gonna die on lol.